pkg/slo-controller/noderesource/framework/extender_plugin.go

```diff
@@ -17,122 +17,138 @@ limitations under the License.
 package framework
 
 import (
-	"fmt"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/klog/v2"
 
 	"github.com/koordinator-sh/koordinator/apis/extension"
 )
 
+const DefaultPluginPriority = 0
+
 var (
-	globalNodePrepareExtender       = map[string]NodePreparePlugin{}
-	globalNodeSyncExtender          = map[string]NodeSyncPlugin{}
-	globalResourceCalculateExtender = map[string]ResourceCalculatePlugin{}
+	globalNodePrepareExtender       = NewRegistry("NodePrepare")
+	globalNodeSyncExtender          = NewRegistry("NodeSync")
+	globalResourceCalculateExtender = NewRegistry("ResourceCalculate")
 )
 
+// Plugin has its name and priority. Registered plugins are executed in order of the priority. The high-priority plugin
+// runs earlier than the low-priority plugin. If the priorities are equal, compare using the lexicographic order.
+type Plugin interface {
+	Name() string
+	Priority() int32
```

Comment:
consider naming like sequence or something

Reply:
removed. just use the order of registration